packages/expo-updates/android/src/main/java/expo/modules/updates/events/UpdatesEventManager.kt

```diff
@@ -1,31 +1,28 @@
 package expo.modules.updates.events
 
-import expo.modules.kotlin.events.EventEmitter
 import expo.modules.updates.logging.UpdatesErrorCode
 import expo.modules.updates.logging.UpdatesLogger
 import expo.modules.updates.statemachine.UpdatesStateContext
+import java.lang.ref.WeakReference
 
 class UpdatesEventManager(private val logger: UpdatesLogger) : IUpdatesEventManager {
-  override var eventEmitter: EventEmitter? = null
+  override var observer: WeakReference<IUpdatesEventManagerObserver>? = null
 
   override fun sendStateMachineContextEvent(context: UpdatesStateContext) {
-    val eventName = UpdatesJSEvent.StateChange.eventName
+    logger.debug("Sending state machine context to observer")
 
-    val eventEmitter = eventEmitter
-    if (eventEmitter == null) {
-      logger.info(
-        "Could not emit $eventName event; no event emitter was found.",
-        UpdatesErrorCode.JSRuntimeError
-      )
+    val observer = observer?.get()
+    if (observer == null) {
+      logger.debug("Unable to send state machine context to observer, no observer", UpdatesErrorCode.JSRuntimeError)
```

Comment:
```suggestion
    val observer = observer?.get() ?: run {
      logger.debug("Unable to send state machine context to observer, no observer", UpdatesErrorCode.JSRuntimeError)
      return
    }
```

nit: kotlin idiom maybe?